Source/Diagnostics/FlushFormats/FlushFormatCheckpoint.cpp

```diff
@@ -159,8 +159,35 @@ FlushFormatCheckpoint::CheckpointParticles (
     const amrex::Vector<ParticleDiag>& particle_diags) const
 {
     for (unsigned i = 0, n = particle_diags.size(); i < n; ++i) {
```

Comment:
In this function and potentially after this look, we probably also need to treat the laser particles to write our their positions

cc @RemiLehe

Reply:
Support for this is added in PR #2360 - I recommend we merge that PR first, then rebase this one.